docs/advanced_usage.rst

```diff
@@ -404,3 +418,52 @@ running Cookiecutter on a template that requires custom Jinja2 extensions.
 .. _`Jinja2 extensions`: http://jinja2.readthedocs.io/en/latest/extensions.html#extensions
 .. _`now`: https://github.com/hackebrot/jinja2-time#now-tag
 .. _`jinja2_time.TimeExtension`: https://github.com/hackebrot/jinja2-time
+
+.. _`running hooks in place`:
+
+Running hooks in place and consuming serialized context
+-------------------------------------------------------
+
+Disabling hook duplication
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+As explained in `Using Pre/Post-Generate Hooks (0.7.0+)`_, you can disable the default behaviour of hook duplication. This is interesting when you don't need to use template variables directly in your hook.
+
+To do so, a template author must specify this wish in ``cookiecutter.json`` as follows:
+
+.. code-block:: json
+
+    {
+        "_no_hookcopy": "yes"
+    }
+
+Using serialized context
+~~~~~~~~~~~~~~~~~~~~~~~~
+Given the ``cookiecutter.json``
+
+.. code-block:: json
+
+    {
+        "project_name": "Cookiecutter example project",
+        "project_slug": "{{ cookiecutter.project_name }}"
+    }
+
+Here follows an example on how to take advantage of the context serialization in ``hooks/pre_gen_project.py``
+
+.. code-block:: python
+
+  #!/usr/bin/env python
+  # -*- coding: utf-8 -*-
+  import sys
+  import json
+  import re
+  
+  # get the serialized context from the standard input
+  context = json.loads(sys.stdin.readlines()[0])
+  
+  # remove 'project' word used in project_name from project_slug
+  context['project_slug'] = re.sub(
+    r'project', '', context['project_slug'], 1, flags=re.I
+  )
+
+  # serialize the updated context and send this modification through the standard output 
+  print(json.dumps(context))
```

Comment:
Your documentation and examples are nice but make this document very long. I suggest moving this and all the hooks docs into a separate hooks.rst for readability.